Source/PINDiskCache.m

```diff
@@ -843,9 +855,10 @@ - (id)objectForKeyedSubscript:(NSString *)key
     
     [self lock];
         BOOL isEmpty = (_dates.count == 0 && _sizes.count == 0);
+        BOOL containsKey = [_knownKeys containsObject:key];
```

Comment:
Here, we are relying on _knownKeys instead of calling containsObjectForKey. Definitely much more efficient, and suggests maybe we can use this as the implementation of the method above.

Reply:
Hopefully my above comment explains why we can't?